scripts/apps/archive/directives/MarkedItemTitle.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @module superdesk.apps.archive
+ * @ngdoc directive
+ * @name sdMarkedItemTitle
+ * @requires desks
+ * @requires authoring
+ * @requires highlightServce
+ * @requires $timeout
+ * @description
+ *   This directive is used in authoring-topbar to mark a story as a highlight or
+ *   to mark it for a desk.
+ */
+MarkedItemTitle.$inject = ['desks', '$timeout', 'authoring', 'highlightsService'];
+export function MarkedItemTitle(desks, $timeout, authoring, highlightsService) {
+    return {
+        scope: {
+            item: '=item',
+            markField: '@field'
+        },
+        templateUrl: 'scripts/apps/archive/views/marked_item_title.html',
+        // todo(petr): refactor to use popover-template once angular-bootstrap 0.13 is out
+        link: function(scope, elem) {
+            /*
+             * Toggle 'open' class on dropdown menu element
+             * @param {string} isOpen
+             */
+            scope.toggleClass = function(isOpen) {
+                scope.open = isOpen;
+            };
+
+            scope.marks = scope.item[scope.markField];
+            scope.hasMarkItemPrivilege = authoring.itemActions(scope.item).mark_item_for_desks;
+
+            scope.$on('item:' + scope.markField, ($event, data) => {
```

Comment:
I didn't look into it that deeply, but if it presents issues I'm happy to help. I think the critical part is breaking this down into smaller parts and documenting it so it's easy to follow. As well as finding an alternative way to introducing the functionality without use of jQuery (if possible) and use of `.click()`.